small/lsregion.c

```diff
@@ -91,3 +91,46 @@ lsregion_aligned_reserve_slow(struct lsregion *lsregion, size_t size,
 	assert((char *)pos + size <= (char *)lslab_end(slab));
 	return pos;
 }
+
+void
+lsregion_to_iovec(const struct lsregion *lsregion, struct iovec *iov,
+		  int *iovcnt, struct lsregion_svp *svp)
+{
+	struct lslab *lslab;
+	int64_t prev_id = svp->max_id;
+	size_t prev_pos = svp->pos;
+	int max_iovcnt = *iovcnt;
+	int cnt = 0;
+	rlist_foreach_entry(lslab, &lsregion->slabs.slabs, next_in_list) {
+		/* An unused slab. */
+		if (lslab->max_id == LSLAB_NOT_USED_ID)
+			continue;
+		/* An already seen slab with no new allocations. */
+		if (lslab->max_id <= prev_id)
+			continue;
+		if (cnt >= max_iovcnt)
+			break;
```

Comment:
Having this check here means that after you filled the iovecs, you still iterate one more time just to break at this place. It seems you could also have the check right after `cnt++` below, or? Then you exit as soon as the iovec array is full.

Reply:
I'd have to handle the case with `max_iovcnt == 0` separately then. That's why I've put the check here.